added ZetaChain Athens testnet

CharlieMc0 opened the pull request:
Add support for the ZetaChain Athens testnet

ZetaChain being added to [ethereum-lists/chains](https://github.com/ethereum-lists/chains) is in progress - https://github.com/ethereum-lists/chains/pull/1963

CharlieMc0 commented:
The CI is failing but the tests work fine locally. Any suggestions to fix this?

In case it helps I included the results from running `NEW_CHAIN_ID=7001 npx mocha test/chains/chain-tests.js --exit` locally. 

```
NEW_CHAIN_ID=7001 npx mocha test/chains/chain-tests.js --exit
Environment variable NODE_URL_MAINNET not set!
Environment variable ALCHEMY_ID not set for eth mainnet!
Environment variable NODE_URL_RINKEBY not set!
Environment variable ALCHEMY_ID not set for eth rinkeby!
Environment variable NODE_URL_GOERLI not set!
Environment variable ALCHEMY_ID not set for eth goerli!
Environment variable NODE_URL_SEPOLIA not set!
Environment variable ALCHEMY_ID not set for eth sepolia!
Environment variable ALCHEMY_ID not set for polygon mainnet!
Environment variable ALCHEMY_ID not set for polygon mumbai!
Environment variable ALCHEMY_ID not set for arb mainnet!
Environment variable ALCHEMY_ID not set for arb goerli!
Environment variable ALCHEMY_ID not set for opt mainnet!
Environment variable ALCHEMY_ID not set for opt goerli!
newAddedChainId
7001


  Test Supported Chains
Injector listening on port 5000!
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[POST:VERIFICATION_BY_ADDRESS_FAILED]","address":"0x1f42652a86918fd84E74e066db94E3078d25Dd8D","msg":"Could not find file in repository","time":"2022-11-30T17:53:48.098Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":40,"msg":"IPFS_API not set. Files will not be pinned to IPFS.","time":"2022-11-30T17:53:48.101Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[RECOMPILE]","fileName":"contracts/1_Storage.sol","contractName":"Storage","version":"0.8.7+commit.e28d00a7","verificationId":"16451758487753798","msg":"Recompiling","time":"2022-11-30T17:53:48.139Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":50,"loc":"[VALIDATE_SOLC_PATH]","solcPath":"/tmp/solc-repo/solc-linux-amd64-v0.8.7+commit.e28d00a7","error":"/tmp/solc-repo/solc-linux-amd64-v0.8.7+commit.e28d00a7: /tmp/solc-repo/solc-linux-amd64-v0.8.7+commit.e28d00a7: cannot execute binary file\n","verificationId":"16451758487753798","msg":"","time":"2022-11-30T17:53:48.170Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[RECOMPILE]","version":"0.8.7+commit.e28d00a7","githubSolcURI":"https://github.com/ethereum/solc-bin/raw/gh-pages/linux-amd64/solc-linux-amd64-v0.8.7%2Bcommit.e28d00a7","verificationId":"16451758487753798","msg":"Fetching executable solc from GitHub","time":"2022-11-30T17:53:48.171Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[RECOMPILE]","version":"0.8.7+commit.e28d00a7","githubSolcURI":"https://github.com/ethereum/solc-bin/raw/gh-pages/linux-amd64/solc-linux-amd64-v0.8.7%2Bcommit.e28d00a7","verificationId":"16451758487753798","msg":"Successfully fetched executable solc from GitHub","time":"2022-11-30T17:53:49.719Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":50,"loc":"[VALIDATE_SOLC_PATH]","solcPath":"/tmp/solc-repo/solc-linux-amd64-v0.8.7+commit.e28d00a7","error":"/tmp/solc-repo/solc-linux-amd64-v0.8.7+commit.e28d00a7: /tmp/solc-repo/solc-linux-amd64-v0.8.7+commit.e28d00a7: cannot execute binary file\n","verificationId":"16451758487753798","msg":"","time":"2022-11-30T17:53:50.578Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[GET_SOLC]","target":"/Users/charlie/git/zeta/sourcify/soljson-repo/soljson-v0.8.7+commit.e28d00a7.js","verificationId":"16451758487753798","msg":"Searching for js solc locally","time":"2022-11-30T17:53:50.579Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[GET_SOLC]","version":"v0.8.7+commit.e28d00a7","verificationId":"16451758487753798","msg":"Searching for js solc remotely","time":"2022-11-30T17:53:50.580Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[GET_SOLC]","version":"v0.8.7+commit.e28d00a7","verificationId":"16451758487753798","msg":"Found solc remotely","time":"2022-11-30T17:53:52.333Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[MATCH]","chain":"7001","address":"0x1f42652a86918fd84E74e066db94E3078d25Dd8D","msg":"Retrieving contract bytecode address","time":"2022-11-30T17:53:52.380Z","v":0}
    ✓ should verify a contract on ZetaChain Athens Testnet (7001) (4563ms)
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[POST:VERIFICATION_BY_ADDRESS_FAILED]","address":"0x7fe7d2F60399913e8ECeb8BAe9adDf9B809E8F6d","msg":"Could not find file in repository","time":"2022-11-30T17:53:52.652Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[RECOMPILE]","fileName":"contracts/WithImmutables.sol","contractName":"WithImmutables","version":"0.7.4+commit.3f05b770","verificationId":"8841990661063459","msg":"Recompiling","time":"2022-11-30T17:53:52.653Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":50,"loc":"[VALIDATE_SOLC_PATH]","solcPath":"/tmp/solc-repo/solc-linux-amd64-v0.7.4+commit.3f05b770","error":"/tmp/solc-repo/solc-linux-amd64-v0.7.4+commit.3f05b770: /tmp/solc-repo/solc-linux-amd64-v0.7.4+commit.3f05b770: cannot execute binary file\n","verificationId":"8841990661063459","msg":"","time":"2022-11-30T17:53:52.710Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[RECOMPILE]","version":"0.7.4+commit.3f05b770","githubSolcURI":"https://github.com/ethereum/solc-bin/raw/gh-pages/linux-amd64/solc-linux-amd64-v0.7.4%2Bcommit.3f05b770","verificationId":"8841990661063459","msg":"Fetching executable solc from GitHub","time":"2022-11-30T17:53:52.711Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[RECOMPILE]","version":"0.7.4+commit.3f05b770","githubSolcURI":"https://github.com/ethereum/solc-bin/raw/gh-pages/linux-amd64/solc-linux-amd64-v0.7.4%2Bcommit.3f05b770","verificationId":"8841990661063459","msg":"Successfully fetched executable solc from GitHub","time":"2022-11-30T17:53:53.868Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":50,"loc":"[VALIDATE_SOLC_PATH]","solcPath":"/tmp/solc-repo/solc-linux-amd64-v0.7.4+commit.3f05b770","error":"/tmp/solc-repo/solc-linux-amd64-v0.7.4+commit.3f05b770: /tmp/solc-repo/solc-linux-amd64-v0.7.4+commit.3f05b770: cannot execute binary file\n","verificationId":"8841990661063459","msg":"","time":"2022-11-30T17:53:54.716Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[GET_SOLC]","target":"/Users/charlie/git/zeta/sourcify/soljson-repo/soljson-v0.7.4+commit.3f05b770.js","verificationId":"8841990661063459","msg":"Searching for js solc locally","time":"2022-11-30T17:53:54.717Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[GET_SOLC]","version":"v0.7.4+commit.3f05b770","verificationId":"8841990661063459","msg":"Searching for js solc remotely","time":"2022-11-30T17:53:54.717Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[GET_SOLC]","version":"v0.7.4+commit.3f05b770","verificationId":"8841990661063459","msg":"Found solc remotely","time":"2022-11-30T17:53:56.998Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[MATCH]","chain":"7001","address":"0x7fe7d2F60399913e8ECeb8BAe9adDf9B809E8F6d","msg":"Retrieving contract bytecode address","time":"2022-11-30T17:53:57.035Z","v":0}
{"name":"VerificationService","hostname":"charlie-zetabook.local","pid":20781,"level":30,"loc":"[GET_CREATION_DATA]","chain":"7001","contractAddress":"0x7fe7d2F60399913e8ECeb8BAe9adDf9B809E8F6d","fetchAddress":"https://blockscout.athens2.zetachain.com/address/0x7fe7d2F60399913e8ECeb8BAe9adDf9B809E8F6d/transactions","msg":"Scraping block explorer","time":"2022-11-30T17:53:57.193Z","v":0}
    ✓ should verify a contract with immutables on ZetaChain Athens Testnet (7001) (6499ms)


  2 passing (11s)
```

marcocastignoli commented:
hey @CharlieMc0 thanks for the PR. Please revert all changes to `services/core/src/chains.json`. We'll soon update the file and you can merge from staging the new file.

kuzdogan commented:
Also please remove `https://archive.athens2.zetachain.com/evm` from the `sourcify-chains.ts` "rpc" field. This should ideally received from the `chains.json`. Keep the rpc in the chain object up-to-date

CharlieMc0 commented:
Thanks for the feedback @kuzdogan @marcocastignoli. I've made the requested changes.

marcocastignoli commented:
Can you rebase on staging / merge from staging? we fixed some issues there

marcocastignoli commented:
(I don't have permissions to do that)

kuzdogan commented on services/core/src/sourcify-chains.ts:
```diff
@@ -577,6 +577,14 @@ const sourcifyChains: SourcifyChainsObject = {
+    monitored: false,
+    contractFetchAddress: "https://blockscout.athens2.zetachain.com/" + BLOCKSCOUT_SUFFIX,
+    txRegex: getBlockscoutRegex(),
+    rpc: [],
```
remove this line, it causes no rpcs to be passed from `chains.json`
```suggestion
```

kuzdogan commented:
Yes sorry wasn't aware it's not in chains.json. We'll let you know once we update the chains, then you can rebase or merge from staging.

kuzdogan commented:
Updated with https://github.com/ethereum/sourcify/commit/f8391fdd47b85bdc1acab305f0fb75437586606e can you please rebase or merge from staging? @CharlieMc0

CharlieMc0 commented:
I updated it but now I'm getting the same errors + GitGurdian thinks I've added secrets into the `chains.json` file plus the other errors are still showing up 

`error: 'Contract name: WithImmutables. Cannot fetch creation data via https://blockscout.athens2.zetachain.com/address/0x7fe7d2F60399913e8ECeb8BAe9adDf9B809E8F6d/transactions on chainId 7001 of contract 0x7fe7d2F60399913e8ECeb8BAe9adDf9B809E8F6d'`  despite the fact that URL loads correctly.

kuzdogan commented:
Please ignore the GitGuardian @CharlieMc0, we're looking to why it failed

@Aldekein this seems to be a false positive

marcocastignoli commented:
Sourcify can verify contracts with immutable variables by comparing the creation bytecodes of the recompiled contract and the deployed contract. To get the creation bytecode of the deployed contract, we scrape the block explorers for the transaction that created the contract (that has the creation bytecode as tx.input) as a workaround. 
You can read more here about verification of contracts with Immutables: https://docs.sourcify.dev/docs/immutables/

Using Athens2's rpc (https://api.athens2.zetachain.com/evm) it is not possible to get the transaction containing the creation data.  Athens2 probably has a rally low tx index setting, the transaction was just 14 days ago. We can still merge this PR but we suggest you to increase the tx index so that Sourcify can provide verification for older contracts.

kuzdogan commented:
Hello the verifications on this chain has been failing since couple weeks because the RPCs are not working. 

Please fix the RPCs, otherwise we'll have to remove the verification support for this chain (this won't delete the existing contracts)

https://app.circleci.com/pipelines/github/ethereum/sourcify/5574/workflows/48df71c3-2f6a-447f-9fd4-7508310f41b6/jobs/32835
https://app.circleci.com/pipelines/github/ethereum/sourcify/5531/workflows/89d73ca4-480a-4b03-bfa9-4402a6388c2c/jobs/32488

CharlieMc0 commented:
We've got a pending PR here to update the RPCs - https://github.com/ethereum-lists/chains/pull/4600

kuzdogan commented:
@CharlieMc0 Great, please ping when it's merged

CharlieMc0 commented:
@kuzdogan the PR in https://github.com/ethereum-lists/chains/pull/4600 was merged. Thanks
